zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushMessageSender.java

```diff
@@ -138,18 +140,21 @@ protected void channelRead0(final ChannelHandlerContext ctx, final FullHttpReque
                 return;
             }
 
-            final ByteBuf body = request.content().retain();
-            if (body.readableBytes() <= 0) {
-                sendHttpResponse(ctx, request, NO_CONTENT, userAuth);
-                return;
-            }
-
             if (pushConn.isRateLimited()) {
                 sendHttpResponse(ctx, request, HttpResponseStatus.SERVICE_UNAVAILABLE, userAuth);
                 logRateLimited();
                 return;
             }
 
+            final ByteBuf body = request.content().retain();
+            if (body.readableBytes() <= 0) {
+                sendHttpResponse(ctx, request, NO_CONTENT, userAuth);
+                // Because we are not passing the body to the pushConn (who would normally handle destroying),
+                // we need to release it here.
+                body.release();
```

Comment:
It's better practice to use `ReferenceCountUtil.release()`

Reply:
Nice! Updated.